### ISSUE TITLE
The output result of OpenJ9-JDK8 will vary depending on whether - XX:+CompactStrings is used

### ISSUE THREAD

**Issue by ayuan0828** [eclipse-openj9/openj9#15954]
**Description**:

As shown in the following test case,when OpenJ9-jdk8 (using the CompactStrings parameter) calls the "insert (int index, String str)" function to insert, if the character array assigned to "str" contains "Character. MAX_VALUE", the last part of characters in the result will be discarded.

I don't know the reason for this result. I applied the same parameter to OpenJ9-jdk-11,  but the same result did not appear.

**Environment**：

```
Ubuntu 18.04 x86_64
```

**JVM Information**:
jdk8:

```openjdk version "1.8.0_332-internal"
OpenJDK Runtime Environment (build 1.8.0_332-internal-_2022_02_17_10_23-b00)
Eclipse OpenJ9 VM (build master-39c490eaf, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20220217_000000 (JIT enabled, AOT enabled)
OpenJ9   - 39c490eaf
OMR      - b63287860
JCL      - 301202f49f based on jdk8u332-b02)
```

jdk11:

```openjdk version "11.0.15-internal" 2022-04-19
OpenJDK Runtime Environment (build 11.0.15-internal+0-adhoc..openj9-openjdk-jdk11)
Eclipse OpenJ9 VM (build master-39c490eaf, JRE 11 Linux amd64-64-Bit Compressed References 20220217_000000 (JIT enabled, AOT enabled)
OpenJ9   - 39c490eaf
OMR      - b63287860
JCL      - 24e7209ed4 based on jdk-11.0.15+1)
```

**Testcase**:

```public class Test {
    public class Test {
    public static void main(String[] args) throws Exception {
        char[] stra = { '1',Character.MAX_VALUE};
        StringBuilder sb = new StringBuilder("12345");
        sb.insert(0, stra);
        System.out.println(sb.toString());
    }
}
```

**Output of JDK8(with  -XX:+CompactStrings)**:

```
1?123
```

**Output of JDK8(default)**:

```
1?12345
```

**Output of JDK11(default)/JDK11(with  -XX:+CompactStrings)**: 

```
1?12345
```

**Comment by pshipton** [comment]
@mikezhang1234567890 pls take a look.

**Comment by mikezhang1234567890** [comment]
This one is fixed in `master`, probably by https://github.com/eclipse-openj9/openj9/pull/15618.

Doesn't look like it's in the 0.33 release though.

**Comment by pshipton** [comment]
@mikezhang1234567890 pls confirm it's fixed in 0.35, which contains the mentioned fix. If so we can close this.

Builds:
https://openj9-jenkins.osuosl.org/job/Pipeline-Release-0.35.0-Build8/
https://openj9-jenkins.osuosl.org/job/Pipeline-Release-0.35.0-Build11/

**Comment by mikezhang1234567890** [comment]
Yes, it's fixed in 0.35:
```
$ ./macjdk8/j2sdk-image/bin/java -version
openjdk version "1.8.0_352-internal"
OpenJDK Runtime Environment (build 1.8.0_352-internal-jenkins_2022_09_24_05_02-b00)
Eclipse OpenJ9 VM (build v0.35.0-release-d8f6fe2f5fd, JRE 1.8.0 Mac OS X amd64-64-Bit Compressed References 20220924_42 (JIT enabled, AOT enabled)
OpenJ9   - d8f6fe2f5fd
OMR      - 447afc0f036
JCL      - 330bf23488c based on jdk8u352-b05)

$ ./macjdk8/j2sdk-image/bin/java -XX:+CompactStrings CompStrTest
1￿12345
```

JDK11+ uses the openjdk implementation, so it wasn't affected by this issue.

**Comment by pshipton** [comment]
Closing as fixed in the next release, due out around Oct 21 or in the next week.
